pkg/operator/reports_test.go

```diff
@@ -92,3 +101,217 @@ func TestGetNextReportPeriod(t *testing.T) {
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.ScheduledReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is true and reason is Scheduled return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionTrue, util.ScheduledReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is false and reason is InvalidReport return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.InvalidReportReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is true and reason is InvalidReport return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionTrue, util.InvalidReportReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is false and reason is RunImmediately return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionFalse, util.RunImmediatelyReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+		{
+			name: "when status running is true and reason is RunImmediately return false",
+			report: testhelpers.NewReport(testReportName, testNamespace, testQueryName, reportStart, reportEnd, v1alpha1.ReportStatus{
+				Conditions: []metering.ReportCondition{
+					*util.NewReportCondition(metering.ReportRunning, v1.ConditionTrue, util.RunImmediatelyReason, "mock-msg"),
+				},
+			}, schedule, false),
+			expectFinished: false,
+		},
+	}
+
+	for _, testCase := range testTable {
+		var mockLogger = logrus.New()
+		testCase := testCase
+
+		t.Run(testCase.name, func(t *testing.T) {
+			runningCond := isReportFinished(mockLogger, testCase.report)
+			assert.Equalf(t, runningCond, testCase.expectFinished, "expected the report '%s' would return '%t', but got '%t'", testCase.name, testCase.expectFinished, runningCond)
+		})
+	}
+}
+
+func TestValidateReport(t *testing.T) {
+	const (
+		testNamespace           = "default"
+		testReportName          = "test-report"
+		testQueryName           = "test-query"
+		testGenerationQueryName = "test-GQN"
```

Comment:
To make this more obvious, perhaps change this to:

```suggestion
		testNonexistentQueryName = "does-not-exist"
```